### DISCUSSION TITLE
How to use transition with form being summited from js

### DISCUSSION THREAD

**Discussion by JakovGlavac** [remix-run/remix#1320] [Q&A] [answered]
Hi, I'm trying to submit form from JS.
I have a modal that is not created inside Form.
![image](https://user-images.githubusercontent.com/34971091/147892455-fbb5b364-e967-46e2-b409-0104f5932fa5.png)

I managed to submit form using ref and js:
```tsx
<Form method="post" ref={solvingForm}>
```
```tsx
<Button
  onClick={() => {
    solvingForm.current.submit();
  }}
>
  Da
</Button>;

```
But the problem is that on submit `transition.submission` is not true, and the whole page refreshes.

https://user-images.githubusercontent.com/34971091/147892537-c9e5f06e-b421-46c7-95ff-f8d14bf1958f.mp4

Is there a way to fix this?

**Comment by sairajchouhan** [comment]
The submit you are using `solvingForm.current.submit()` is HTML `form` submission even though you are using remix `Form` because you are accessing it from a `ref`

To submit a remix `Form` programmatically use `useSubmit` hook


```tsx
import { useSubmit } from 'remix'

export default function Component(){
  const submit = useSubmit()

    return (
      <Form method="post">
        <Button
          onClick={() => {
            submit(solvingForm.current, {
             method: 'post',
             action: '/route-you-want-to-post'
            })
          }}
        >
          Da
       </Button>;
    </Form>
  )
}

```

Then you can access all the properties from `useTransition` hook and `transition.submit` will also be `true`

**Reply by JakovGlavac** [reply]
Yeah this is cool, but how to make it work outside of `<Form>` ?

**Comment by sergiodxa** [comment, marked as answer]
The simpler way is to put an ID to the Form and then add a form attribute to the button element with the ID of the form. That will tell the browser the form of that button is your other form and when you click it (if type is submit) it will submit the form.

This way you don’t need to grab a ref or use the useSubmit hook, also it will work even without JS.

**Reply by sergiodxa** [reply]
Btw this is the attribute https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#attr-form

**Reply by sairajchouhan** [reply]
Wow that's very helpful, thanks @sergiodxa
